src-tauri/src/main.rs

```diff
@@ -1,8 +1,56 @@
 // Prevents additional console window on Windows in release, DO NOT REMOVE!!
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
 
+use std::{fs::DirEntry, io::{Error, Result}, path::{Path, PathBuf}};
+use tauri::api::path::home_dir;
+
 fn main() {
   tauri::Builder::default()
+    .invoke_handler(tauri::generate_handler![
+      list
+      ])
     .run(tauri::generate_context!())
     .expect("error while running tauri application");
 }
+
+/// list all directories that have a `README.md` file inside
```

Comment:
I would like to have a documentation of the intended purpose of the function rather than it's implementation, since I can see the implementation just from the code. I would perfer something like this:

```suggestion
/// list this names of the notes, which is the list of all directories in the base directory  that have a `README.md` file inside
```